internal/common/validation/job.go

```diff
@@ -5,15 +5,95 @@ import (
 
 	"github.com/G-Research/armada/internal/armada/configuration"
 	"github.com/G-Research/armada/internal/common/armadaerrors"
+	"github.com/G-Research/armada/internal/common/util"
+	"github.com/G-Research/armada/internal/scheduler"
 
 	"github.com/G-Research/armada/pkg/api"
 )
 
-func ValidateApiJob(j *api.Job, config configuration.PreemptionConfig) error {
-	if err := ValidateApiJobPodSpecs(j); err != nil {
+func ValidateApiJobs(jobs []*api.Job, config configuration.SchedulingConfig) error {
+	err := validateGangs(jobs, config.GangIdAnnotation, config.GangCardinalityAnnotation)
+	if err != nil {
 		return err
 	}
-	return ValidatePodSpecPriorityClass(j.PodSpec, config.Enabled, config.PriorityClasses)
+	for _, job := range jobs {
+		if err := ValidateApiJob(job, config); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func validateGangs(jobs []*api.Job, gangIdAnnotation, gangCardinalityAnnotation string) error {
```

Comment:
we probably also have to validate that tolerations, node affinities and node selectors are equal for everything in the gang.  Otherwise you could end up with a gang that could never be scheduled (specifically the case where all members of the gang can be scheduled,  but only only disjoint clusters.)

Reply:
My preference is to leave it as-is for now and to add a proper check that runs the scheduler against the job in a subsequent PR.

Reply:
We need a way to run the `NodeDb` matching logic in a special mode that doesn't consider resources consumed by running jobs. If we then also cache the most recent `NodeDb` for each cluster, we can run submitted jobs against these cached `NodeDb` instances and check if any could schedule the job.

We'd need to do this anyway if we are to keep a submission-time check when we remove core Redis and the old scheduler.